core/codec/cbor/cbor.hpp

```diff
@@ -18,10 +18,14 @@ namespace fc::codec::cbor {
    * @return encoded data
    */
   template <typename T>
-  std::vector<uint8_t> encode(const T &arg) {
-    CborEncodeStream encoder;
-    encoder << arg;
-    return encoder.data();
+  outcome::result<std::vector<uint8_t>> encode(const T &arg) {
```

Comment:
In order to provide this operator with ability to work with multiple arguments you can consult with scale implementation [here](https://github.com/soramitsu/kagome/blob/526ccc4766bbd8dd1b660be10e33ff730c00e710/core/scale/scale.hpp#L26)

Reply:
Filecoin uses IPLD, where each CBOR encoded object must have single root element.
I think it's better to leave it with single argument.